test/ffmpeg-vaapi/transcode/transcoder.py

```diff
@@ -10,35 +10,135 @@
 
 @slash.requires(have_ffmpeg)
 @slash.requires(have_ffmpeg_vaapi_accel)
+@platform_tags(ALL_PLATFORMS)
 class TranscoderTest(slash.Test):
+  requirements = dict(
+    decode = {
+      "avc" : dict(
+        sw = (ALL_PLATFORMS, have_ffmpeg_decoder("h264"), None),
+        # for ffmpeg-vaapi HW decoders are built-in and can't be validated
+        hw = (AVC_DECODE_PLATFORMS, (True, True), None),
+      ),
+      "hevc-8" : dict(
+        sw = (ALL_PLATFORMS, have_ffmpeg_decoder("hevc"), None),
+        # for ffmpeg-vaapi HW decoders are built-in and can't be validated
+        hw = (HEVC_DECODE_8BIT_PLATFORMS, (True, True), None),
+      ),
+      "mpeg2" : dict(
+        hw = (MPEG2_DECODE_PLATFORMS, (True, True), None),
+      ),
+      "vc1" : dict(
+        hw = (VC1_DECODE_PLATFORMS, (True, True), None),
+      ),
+    },
+    encode = {
+      "avc" : dict(
+        sw = (ALL_PLATFORMS, have_ffmpeg_encoder("libx264"), ("libx264", "h264")),
+        hw = (AVC_ENCODE_PLATFORMS, have_ffmpeg_encoder("h264_vaapi"), ("h264_vaapi", "h264")),
+      ),
+      "hevc-8" : dict(
+        sw = (ALL_PLATFORMS, have_ffmpeg_encoder("libx265"), ("libx265", "h265")),
+        hw = (HEVC_ENCODE_8BIT_PLATFORMS, have_ffmpeg_encoder("hevc_vaapi"), ("hevc_vaapi", "h265")),
+      ),
+      "mpeg2" : dict(
+        sw = (ALL_PLATFORMS, have_ffmpeg_encoder("mpeg2video"), ("mpeg2video", "m2v")),
+        hw = (MPEG2_ENCODE_PLATFORMS, have_ffmpeg_encoder("mpeg2_vaapi"), ("mpeg2_vaapi", "m2v")),
+      ),
+      "mjpeg" : dict(
+        sw = (ALL_PLATFORMS, have_ffmpeg_encoder("mjpeg"), ("mjpeg", "mjpeg")),
+        hw = (JPEG_ENCODE_PLATFORMS, have_ffmpeg_encoder("mjpeg_vaapi"), ("mjpeg_vaapi", "mjpeg")),
+      ),
+    }
+  )
+
+  # hevc implies hevc 8 bit
+  requirements["encode"]["hevc"] = requirements["encode"]["hevc-8"]
+  requirements["decode"]["hevc"] = requirements["decode"]["hevc-8"]
+
   def before(self):
     self.refctx = []
+ 
+  def get_requirements_data(self, ttype, codec, mode):
+      plats, reqs, ffparams = self.requirements[ttype].get(
```

Comment:
Assignment to local variables is unnecessary.  Just do `return self.requirements...`

Reply:
modified it